### PR TITLE
Add login and complete authentication system

### PR DESCRIPTION
**Pull request by tojawawel**

### CODE REVIEW

File: app/controllers/application_controller.rb [hunk 1]
```diff
@@ -1,2 +1,19 @@
 class ApplicationController < ActionController::Base
+
+  helper_method :current_user, :logged_in?
+
+  def current_user
+    @current_user ||= User.find(session[:user_id]) if session[:user_id]
+  end
+
+  def logged_in?
+    !!current_user
```
Comment (Maciorex) [inline app/controllers/application_controller.rb]: double `!!`

Comment (tojawawel) [reply app/controllers/application_controller.rb]: it's supposed to be double

Comment (Maciorex) [reply app/controllers/application_controller.rb]: ok, but why?

Comment (tojawawel) [reply app/controllers/application_controller.rb]: it's converting into proper boolean value

**Pull request merged**